src/map/unit.hpp

```diff
@@ -64,6 +64,20 @@ struct unit_data {
 	int32 group_id;
 
 	std::vector<int32> shadow_scar_timer;
+
+	// Functions and struct to calculate and store exact position at a certain tick
+	int16 getx(t_tick tick = gettick());
+	int16 gety(t_tick tick = gettick());
+	void getpos(int16 &x, int16 &y, uint8 &sx, uint8 &sy, t_tick tick = gettick());
```

Comment:
Why is it necessary to hand over the tick from the external caller?

Reply:
Then we should not use default values - that go against this principle - actually.

Reply:
Some functions don't have a tick, so I wanted to make the call shorter for those cases.

I was wondering if we could use macros for making it shorter instead though.

Maybe I'm overthinking, my logic was that `bl->x` is just 5 characters and I didn't want to turn getting coordinates this way into like 20 characters.